### PR TITLE
O3- 2998 monthly calender in appointment form fixed

### PR DESCRIPTION
**Pull request by Madhu-mac**
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-docs.openmrs.org/docs/frontend-modules/contributing.en-US#contributing-guidelines) label. See existing PR titles for inspiration.
- [x] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://om.rs/o3ui).
- [x] My work includes tests or is validated by existing tests.

## Summary
 This PR Fixes the issue with new enhanced monthly calender view 

- In this new enhanced monthly calendar view the user can able to go to the future months with the arrow buttons.
- And it only lets the user to select current and future dates which is the actual requirement needed here. The user should not 
select past dates. 
- The selected date gets highlighted in the calendar for better experience

## Screenshots
 

https://github.com/openmrs/openmrs-esm-patient-management/assets/99901964/454006d5-88a3-43c9-ae28-2afcad7c382d

## Related Issue
https://github.com/openmrs/openmrs-esm-patient-management/pull/1049

### CODE REVIEW

**Comment by Madhu-mac** [comment]
@mogoodrich Can you please review this PR

**Comment by mogoodrich** [comment]
I believe @gabriel090  may already be looking at this, I will defer to @gabriel090  for the initial review.

**Comment by Madhu-mac** [comment]
> I believe @gabriel090 may already be looking at this, I will defer to @gabriel090 for the initial review.

Sure! Hope to see a good feedback from the review

**Comment by ojwanganto** [comment]
@mogoodrich I see this PR adds the extra navigation that @gabriel090 may not be working on. I like his addition

**Comment by Madhu-mac** [comment]
@ojwanganto I've resolved the conflicts. Can you review this..

**Comment by mogoodrich** [comment]
@ojwanganto  @Madhu-mac  I'm a bit "underwater" right now with some things, but should be able to check out later in the week.

**Comment by Madhu-mac** [comment]
> @ojwanganto  @Madhu-mac  I'm a bit "underwater" right now with some things, but should be able to check out later in the week.

Cool

**Review by mogoodrich — COMMENTED** [review]
It looks like the dates that can't be clicked on are black and the ones that are clickable are grey. Can we reverse so that the future dates are black and the dates in the past are grey?

Also, I think we this PR includes some other changes from other PRs, would be it be possible to have a clean PR with just the changes that are part of this PR?

**Comment by Madhu-mac** [comment]
> It looks like the dates that can't be clicked on are black and the ones that are clickable are grey. Can we reverse so that the future dates are black and the dates in the past are grey?
> 
> Also, I think we this PR includes some other changes from other PRs, would be it be possible to have a clean PR with just the changes that are part of this PR?

Yeah! I like the idea as well. will be working on it

**Comment by gabriel090** [comment]
> It looks like the dates that can't be clicked on are black and the ones that are clickable are grey. Can we reverse so that the future dates are black and the dates in the past are grey?
> 
> Also, I think we this PR includes some other changes from other PRs, would be it be possible to have a clean PR with just the changes that are part of this PR?

Some changes and modifications have been worked on. We can select the date from the grid and save the appointment on the selected date. Open for feedback Thank you.
appointmentSchedular

**Review by mogoodrich — APPROVED** [review]
LGTM, thanks @Madhu-mac .... but looks like there are some conflicts to resolve before merging?

**Comment by Madhu-mac** [comment]
@mogoodrich I've resolved the confilcts and tested it locally it works fine..!!

**Comment by mogoodrich** [comment]
@Madhu-mac  are these latest changes ready for review? can you explain what the updates do?

**Comment by Madhu-mac** [comment]
> @Madhu-mac are these latest changes ready for review? can you explain what the updates do?

Yea its ready for the review & the deliverables are..
- The code defines a React functional component called MonthlyCalendarView that renders a monthly calendar view.
- It manages the state of the currently displayed month and the selected date using the useState hook.
- The component includes event handlers for navigating between months and selecting dates, renders the calendar UI with workload counts

**Comment by mogoodrich** [comment]
Thanks @Madhu-mac !  Can you look into the failing tests and once we fix those we likely can merge this in?

**Comment by mogoodrich** [comment]
@Madhu-mac  you should be able to see the failing tests below...

**Comment by Madhu-mac** [comment]
@mogoodrich I've added some changes & the test cases are running fine. Can you look into it..

File: packages/esm-appointments-app/src/form/appointments-form.component.tsx
Comment (mogoodrich) [inline packages/esm-appointments-app/src/form/appointments-form.component.tsx:355]: Where is this used?

File: yarn.lock
Comment (mogoodrich) [inline yarn.lock:6754]: is this update necessary?

File: packages/esm-appointments-app/src/form/appointments-form.component.tsx
Comment (mogoodrich) [inline packages/esm-appointments-app/src/form/appointments-form.component.tsx:487]: I see a bunch of references to reoccurring dates in the old code, did we confirm that scheduling reoccurring appointments still works after these changes?

**Review by mogoodrich — COMMENTED** [review]
Thanks!  I am running the tests now, but, I also just added some questions I have.

File: packages/esm-appointments-app/src/form/appointments-form.component.tsx
Comment (Madhu-mac) [reply packages/esm-appointments-app/src/form/appointments-form.component.tsx:362]: The `updateLocations` function is defined in the AppointmentsForm component but is not directly used within the component itself. We can use it to optimize performance by fetching and updating the list only when necessary, rather than fetching it every time the component renders.( Its for the future purposes if its relative static)

File: yarn.lock
Comment (Madhu-mac) [reply yarn.lock:6754]: Depends, If we wanna cope up with new versions or we can just revert it. Its working same if we update the version also not much of a problem

File: packages/esm-appointments-app/src/form/appointments-form.component.tsx
Comment (Madhu-mac) [reply packages/esm-appointments-app/src/form/appointments-form.component.tsx:494]: Updating this value as intended doesn't cause any issues with the handling of recurring appointment end dates. Its working fine..

**Comment by Madhu-mac** [comment]
@mogoodrich Is there anything that i need to look into?..

**Comment by Madhu-mac** [comment]
@mogoodrich I've commented out `updateLocations` fn las you said.. Do i need to make any changes to this PR bfr Merging to the main

**Comment by mogoodrich** [comment]
Thanks @Madhu-mac and sorry for the delay... assuming the tests pass (I just triggered them to re-run) this is good to merge, thanks!

**Comment by Madhu-mac** [comment]
> Thanks @Madhu-mac and sorry for the delay... assuming the tests pass (I just triggered them to re-run) this is good to merge, thanks!

Thanks for the assist @mogoodrich

**Comment by mogoodrich** [comment]
No worries, @Madhu-mac , let me know if you need anything else from me to get this merged.

**Comment by mogoodrich** [comment]
Unfortunately I am reverting this PR, because as @mccarthyaaron  mentioned (see link above) this commit removed the end date picker for reoccurring appointments and therefore broke that functionality:

https://github.com/openmrs/openmrs-esm-patient-management/commit/47663ad6618b3384ce0fe44a50ff39c259c00f80

**Pull request merged**
